src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -165,10 +163,15 @@ createFxAdjustedPrincipalClaim dateToTime ownerReceives fxRateMultiplier notiona
     prepareAndTagClaims dateToTime [fxLegClaim] "Principal payment"
 -- FIXED_RATE_BOND_REDEMPTION_CLAIM_END
 
--- | Scale a claim by an asset's performance. Specifically, this multiplies the input claim by the quantity `S(endDate) / S(startDate) - 1.0`
+-- | Scale a claim by an asset's performance. Specifically, this multiplies the input claim by the
+-- quantity `S(endDate) / S(startDate) - 1.0`
 scaleByAssetPerformance : Date -> Date -> O -> C -> C
 scaleByAssetPerformance startDate endDate asset claim =
-  when (TimeGte startDate) $ scale (Const 1.0 / asset) $
-  when (TimeGte endDate) $ scale (asset) $
-  claim
-
+  mconcat $ [perfClaim, correctionClaim]
```

Comment:
Just fyi, this can also be written as

```suggestion
  perfClaim `and` correctionClaim
```

Reply:
I think that looks a bit nicer, but it caused a compiler error, so I reverted back to `mconcat`.

Reply:
Interesting, what was the error?

Reply:
In order to remove the error, you need to `import Prelude hiding (and)` and probably import the `and` builder from claims

Reply:
Thanks, that did the trick.